src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -1766,6 +1787,49 @@ FindAndMoveShardCost(float4 utilizationLowerBound, float4 utilizationUpperBound,
 					 */
 					continue;
 				}
+
+				/*
+				 * fmaxf and fminf here are only needed for cases when nodes
+				 * have different capacities. If they are the same, then both
+				 * arguments are equal.
+				 */
+				float4 utilizationImprovement = fmaxf(
+					sourceFillState->utilization - newTargetUtilization,
+					newSourceUtilization - targetFillState->utilization
+					);
+				float4 utilizationAddedByShard = fminf(
+					newTargetUtilization - targetFillState->utilization,
+					sourceFillState->utilization - newSourceUtilization
+					);
+
+				/*
+				 * If the shard causes a lot of utilization, but the
+				 * improvement is gained by moving it is small then we ignore
+				 * it. Probably there are other shards that are better
+				 * candidates, and in any case it's probably not worth the
+				 * effort to move the this shard.
+				 *
+				 * One of the main cases this tries to avoid is the rebalancer
+				 * moving a very large shard with the "by_disk_size" strategy
+				 * when that only gives a small benefit in data distribution.
+				 */
+				if (utilizationImprovement < utilizationAddedByShard *
+					utilizationImproventThreshold)
+				{
+					ereport(NOTICE, (
+								errmsg(
+									"Ignoring move of shard %ld from %s:%d to %s:%d, because it is a big shard and the move only brings a small improvement",
+									shardCost->shardId,
+									sourceFillState->node->workerName,
+									sourceFillState->node->workerPort,
+									targetFillState->node->workerName,
+									targetFillState->node->workerPort
+									),
+								errhint(
+									"If you do want this move to happen try changing utilization_improvement_threshold to a lower value, such as 0.2")
+								));
```

Comment:
due to the recursive behaviour of the planner my expectation is that this message might show quite often for the same moves. Is there any way where we can prevent the same move from showing repeaditly?

Also, there might be many different moves that hit here. Could we also prevent more than x (say 5 or 10) of these messages showing? Ideally we would summarise this at the end with something like

```
Ignored %d moves possibilities, shown %d occurrences in logs.
```

This could provide the user with some ideas if their cutoff is well tuned or not. We will need to see if these messages adds much information that is actionable to the user. If not, we might want to keep them only as `DEBUG` messages.